src/extension.ts

```diff
@@ -499,6 +500,72 @@ function launchMetals(
       });
     });
 
+    registerCommand("metals.new-scala-file", (dirArg: Uri) => {
+      const directory = (() => {
+        if (dirArg !== undefined) {
+          return dirArg.toString();
+        } else if (window.activeTextEditor) {
+          const currentPath = window.activeTextEditor.document.uri.fsPath;
+          return Uri.file(path.dirname(currentPath)).toString();
+        }
+        // NOTE(aleksei): I would return workspace folder here, but not sure if there's API for that.
+        // Moving this funcitonality to server.
+        return undefined;
```

Comment:
moving stuff to the server makes sense, but I would change this note to something more "final". 

Also, what API are you looking for? `workspace.workspaceFolders[0]` should give you what you want